andela/bp-esa-backend
# #163289705 Skip duplicate automation

#### What does this PR do?
Checks that new placementId has no duplicate in the database and create or skip new automation subsequently

#### Description of Task to be completed?
- Add a unique placementId column to identify duplicate placements
- Before executing an automation, check if same placementId exists from a previously executed automation. Skip automation if exists, else proceed with automation

#### How should this be manually tested?
1. On terminal, run: `npx sequelize db:migrate:undo:all` to undo migrations
2. Run: `npx sequelize db:migrate` to execute migration which adds new placementId column to automations table
3. On terminal, run command: `npm start`
4. Set `TIME_INTERVAL` to a short duration like 10 seconds
5. Create new placement using mockAlocations endpoint and wait for automation to execute
6. Observe that the automation was not executed twice for the same data even though `fetchNewPlacement` function fetches it repeatedly, being a placement within the past `24hours`

#### Any background context you want to provide?
When a new placement is fetched from allocations api, there is a risk that it has been fetched before and therefore create a duplicate record on ESA database

#### What are the relevant pivotal tracker stories?
[#163289705](https://www.pivotaltracker.com/story/show/163289705)
